### PR TITLE
Add support to marketplace.

### PR DESCRIPTION
**Pull request by Bonafede**

### CODE REVIEW

File: src/CheckoutSdk/CheckoutConfiguration.cs [hunk 1]
```diff
@@ -1,14 +1,39 @@
...
+        public CheckoutFilesConfiguration GetFilesApiConfiguration()
+        {
+            return FilesApiConfiguration;
+        }
+
+        private void Init(
+            SdkCredentials sdkCredentials,
+            Environment environment,
             IHttpClientFactory httpClientFactory)
         {
```
Comment (rsaestrela) [inline src/CheckoutSdk/CheckoutConfiguration.cs:38]: Method `Init` is not needed. Just assign the variables in the constructor and remove the redundant `private set;`s

File: src/CheckoutSdk/Marketplace/Company.cs [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        public string LegalName { get; set; }
+
+        [JsonProperty(PropertyName = "trading_name")]
+        public string TradingName { get; set; }
+
+        [JsonProperty(PropertyName = "principal_address")]
+        public Address PrincipalAddress { get; set; }
+
+        [JsonProperty(PropertyName = "registered_address")]
+        public Address RegisteredAddress { get; set; }
```
Comment (rsaestrela) [inline src/CheckoutSdk/Marketplace/Company.cs:22]: Please remove any `[JsonProperty]` that are not needed. Just to be clear, the serializer automatically converts from CamelCase to Snake Case (AbcDef to abc_def). Just need to add these when there are exceptions.

File: src/CheckoutSdk/Marketplace/DateOfBirth.cs [hunk 3]
```diff
@@ -0,0 +1,11 @@
+﻿namespace Checkout.Marketplace
+{
+    public class DateOfBirth
+    {
+        public int Day { get; set; }
```
Comment (rsaestrela) [inline src/CheckoutSdk/Marketplace/DateOfBirth.cs:5]: Please add ?.

File: src/CheckoutSdk/Marketplace/MarketplaceClient.cs [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+        public async Task<OnboardEntityResponse> UpdateEntity(OnboardEntityRequest entityRequest, string entityId)
+        {
+            CheckoutUtils.ValidateParams("entityRequest", entityRequest, "entityId", entityId);
+            return await ApiClient.Put<OnboardEntityResponse>(BuildPath(MarketplacePath, EntitiesPath, entityId), SdkAuthorization(), entityRequest);
+        }
+
+        public async Task CreatePaymentInstrument(MarketplacePaymentInstrument marketplacePaymentInstrument, string entityId)
+        {
+            CheckoutUtils.ValidateParams("marketplacePaymentInstrument", marketplacePaymentInstrument, "entityId", entityId);
+            await ApiClient.Post<bool>(BuildPath(MarketplacePath, EntitiesPath, entityId, InstrumentPath), SdkAuthorization(), marketplacePaymentInstrument);
```
Comment (rsaestrela) [inline src/CheckoutSdk/Marketplace/MarketplaceClient.cs:47]: Should we return `bool` is this case? What was the solution found for this use cases? @Armando-Ibarra

File: src/CheckoutSdk/Marketplace/MarketplaceFilePurpose.cs [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    public class MarketplaceFilePurpose
+    {
+        private MarketplaceFilePurpose(string purpose)
+        {
+            Value = purpose;
+        }
+
+        public string Value { get; private set; }
+
+        public static MarketplaceFilePurpose BankVerification
```
Comment (rsaestrela) [inline src/CheckoutSdk/Marketplace/MarketplaceFilePurpose.cs:12]: Nice stuff!

File: src/CheckoutSdk/Marketplace/MarketplacePaymentInstrument.cs [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+
+        public CountryCode Country { get; set; }
+
+        public InstrumentDocument Document { get; set; }
+
+        public BankDetails Bank { get; set; }
+
+        public AccountHolder AccountHolder { get; set; }
+
+        public InstrumentType Type
```
Comment (rsaestrela) [inline src/CheckoutSdk/Marketplace/MarketplacePaymentInstrument.cs:38]: Can we replace this with: `public readonly InstrumentType Type = InstrumentType.BankAccount;`?

File: src/CheckoutSdk/Marketplace/OnboardingStatus.cs [hunk 7]
```diff
@@ -0,0 +1,12 @@
+﻿namespace Checkout.Marketplace
+{
+    public enum OnboardingStatus
```
Comment (rsaestrela) [inline src/CheckoutSdk/Marketplace/OnboardingStatus.cs:3]: 1. Please use .NET enum member naming convention (Camel Case)
2. As it is this shouldn't work - `Please add [EnumMember(Value = "value")]` on each entry so it's correctly serialized/desirialized.

**Review by rsaestrela — CHANGES_REQUESTED** [review]

File: src/CheckoutSdk/Marketplace/MarketplaceClient.cs (hunk 4 above)
Comment (Armando-Ibarra) [reply src/CheckoutSdk/Marketplace/MarketplaceClient.cs:47]: I think that the actual response is an `Id` said that `IdResponse` should be enough.

File: src/CheckoutSdk/Marketplace/MarketplacePaymentInstrument.cs [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+
+        public string BranchCode { get; set; }
+
+        public string Iban { get; set; }
+
+        public string Bban { get; set; }
+
+        public string SwiftBic { get; set; }
+
+        public Currency Currency { get; set; }
```
Comment (AleGladchteinParser) [inline src/CheckoutSdk/Marketplace/MarketplacePaymentInstrument.cs:28]: Maybe this field could be null?

File: src/CheckoutSdk/Marketplace/MarketplacePaymentInstrument.cs [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+
+namespace Checkout.Marketplace
+{
+    public class MarketplacePaymentInstrument
+    {
+        public readonly InstrumentType Type = InstrumentType.BankAccount;
+
+        public string Label { get; set; }
+
+        public AccountType AccountType { get; set; }
```
Comment (Armando-Ibarra) [inline src/CheckoutSdk/Marketplace/MarketplacePaymentInstrument.cs:14]: Please add ?

File: src/CheckoutSdk/Marketplace/MarketplacePaymentInstrument.cs (hunk 8 above)
Comment (Armando-Ibarra) [inline src/CheckoutSdk/Marketplace/MarketplacePaymentInstrument.cs:28]: Please add ?

File: src/CheckoutSdk/Marketplace/MarketplacePaymentInstrument.cs [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+
+        public string Iban { get; set; }
+
+        public string Bban { get; set; }
+
+        public string SwiftBic { get; set; }
+
+        public Currency Currency { get; set; }
+
+        public CountryCode Country { get; set; }
```
Comment (Armando-Ibarra) [inline src/CheckoutSdk/Marketplace/MarketplacePaymentInstrument.cs:30]: Same as previous

File: src/CheckoutSdk/Marketplace/OnboardEntityDetailsResponse.cs [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+
+        public string Reference { get; set; }
+
+        public Capabilities Capabilities { get; set; }
+
+        public string Status { get; set; }
+
+        public IList<RequirementsDue> RequirementsDue { get; set; }
+
+        [JsonProperty(PropertyName = "contact_details")]
```
Comment (Armando-Ibarra) [inline src/CheckoutSdk/Marketplace/OnboardEntityDetailsResponse.cs:19]: This is not needed

File: src/CheckoutSdk/Marketplace/MarketplacePaymentInstrument.cs [hunk 12]
```diff
@@ -0,0 +1,38 @@
+﻿using Checkout.Common;
+using Checkout.Common.Four;
+using Checkout.Instruments;
+using Newtonsoft.Json;
+
+namespace Checkout.Marketplace
+{
+    public class MarketplacePaymentInstrument
```
Comment (Armando-Ibarra) [inline src/CheckoutSdk/Marketplace/MarketplacePaymentInstrument.cs:8]: Please add ? for all Enums in this class.

File: src/CheckoutSdk/Marketplace/OnboardEntityRequest.cs [hunk 13]
```diff
@@ -0,0 +1,18 @@
+﻿using Newtonsoft.Json;
+
+namespace Checkout.Marketplace
+{
+    public class OnboardEntityRequest
+    {
+        public string Reference { get; set; }
+
+        [JsonProperty(PropertyName = "contact_details")]
```
Comment (Armando-Ibarra) [inline src/CheckoutSdk/Marketplace/OnboardEntityRequest.cs:9]: Not needed.

File: src/CheckoutSdk/Marketplace/OnboardEntityResponse.cs [hunk 14]
```diff
@@ -0,0 +1,20 @@
...
+    {
+        public string Id { get; set; }
+
+        public string Reference { get; set; }
+
+        public Capabilities Capabilities { get; set; }
+
+        public OnboardingStatus Status { get; set; }
+
+        [JsonProperty(PropertyName = "requirements_due")]
```
Comment (Armando-Ibarra) [inline src/CheckoutSdk/Marketplace/OnboardEntityResponse.cs:17]: Not needed.

File: src/CheckoutSdk/Marketplace/Representative.cs [hunk 15]
```diff
@@ -0,0 +1,24 @@
+﻿using Checkout.Common;
+using Newtonsoft.Json;
+
+namespace Checkout.Marketplace
+{
+    public class Representative
+    {
+        public string Id { get; set; }
+
+        [JsonProperty(PropertyName = "first_name")]
```
Comment (Armando-Ibarra) [inline src/CheckoutSdk/Marketplace/Representative.cs:10]: This is not needed

File: src/CheckoutSdk/Marketplace/Representative.cs [hunk 16]
```diff
@@ -0,0 +1,24 @@
...
+        public string FirstName { get; set; }
+
+        [JsonProperty(PropertyName = "last_name")]
```
Comment (Armando-Ibarra) [inline src/CheckoutSdk/Marketplace/Representative.cs:13]: Same as previous.

File: src/CheckoutSdk/Marketplace/Representative.cs [hunk 17]
```diff
@@ -0,0 +1,24 @@
+﻿using Checkout.Common;
+using Newtonsoft.Json;
+
+namespace Checkout.Marketplace
+{
+    public class Representative
```
Comment (Armando-Ibarra) [inline src/CheckoutSdk/Marketplace/Representative.cs:6]: Please add ?  for all Enums in this class.

**Review by Armando-Ibarra — CHANGES_REQUESTED** [review]
Consider the following.

The JsonProperty property members are only required for special scenarios. By default the serialization can handle camel case and underscore for properties with lower case.

Add ? for Enums, int, Dates, and any other member that by default uses the position 0.

Let's review the implementation of FilesClient together I have some concerns.

**Pull request closed without merging**